.github/workflows/ci.yaml

```
@@ -76,20 +43,28 @@ jobs:
+    permissions:
+      contents: write # needed for mutex
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v4.2.2
+        uses: actions/checkout@v5.0.0
+        with:
+          persist-credentials: false
 
       - name: Move python script that replaces private package registry information to temp folder so it doesn't get deleted
         run: |
```

actionlint (1.7.9)

shellcheck reported issue in this script: SC2086:info:1:60: Double quote to prevent globbing and word splitting

(shellcheck)